src/components/NcButton/NcButton.vue

```diff
@@ -196,6 +196,66 @@ button {
 }
 </style>
 ```
+
+### Pressed state
+It is possible to make the button stateful by adding a pressed state, e.g. if you like to create a favorite button.
+The button will have the required `aria` attribute for accessibility and visual style (`primary` when pressed, and the configured type otherwise).
+
+```vue
+<template>
+	<div>
+		<div style="display: flex; gap: 12px;">
+			<NcButton :pressed="isFavorite" :aria-label="ariaLabel" type="tertiary-no-background" @click="toggleFavorite">
+				<template #icon>
+					<IconStar v-if="isFavorite" :size="20" />
+					<IconStarOutline v-else :size="20" />
+				</template>
+			</NcButton>
+			<NcButton :pressed="isFavorite" :aria-label="ariaLabel" type="tertiary" @click="toggleFavorite">
+				<template #icon>
+					<IconStar v-if="isFavorite" :size="20" />
+					<IconStarOutline v-else :size="20" />
+				</template>
+			</NcButton>
+			<NcButton :pressed="isFavorite" :aria-label="ariaLabel" @click="toggleFavorite">
+				<template #icon>
+					<IconStar v-if="isFavorite" :size="20" />
+					<IconStarOutline v-else :size="20" />
+				</template>
+			</NcButton>
+		</div>
+		<div>
+			It is {{ isFavorite ? 'a' : 'not a' }} favorite.
+		</div>
+	</div>
+</template>
+<script>
+import IconStar from 'vue-material-design-icons/Star.vue'
+import IconStarOutline from 'vue-material-design-icons/StarOutline.vue'
+
+export default {
+	components: {
+		IconStar,
+		IconStarOutline,
+	},
+	data() {
+		return {
+			isFavorite: false,
+		}
+	},
+	computed: {
+		ariaLabel() {
+			return this.isFavorite ? 'Remove as favorite' : 'Add as favorite'
+		},
+	},
+	methods: {
+		toggleFavorite() {
+			this.isFavorite = !this.isFavorite
+		},
+	},
```

Comment:
What about adding `update:pressed` event to the component?
Then `pressed` state will be synced and component's user won't need to handle clicks manually.